backend/src/review/review.service.spec.ts

```diff
@@ -0,0 +1,138 @@
+
+      const similarity = service.reviewSimilarity(review1, review2);
+
+      expect(similarity).toBe(expectedSimilarity);
+    });
+  });
+
+  describe('checkReviewsForSimilarity', () => {
+    it('should return true if a similar review is found in the provided array', async () => {
+      const reviewsFromDb: Review[] = [
+        {
+          id: 1,
+          landlord: 'John Doe',
+          country_code: 'US',
+          city: 'New York',
+          state: 'NY',
+          zip: '10001',
+          review: 'This is a great product!',
+          repair: 4,
+          health: 3,
+          stability: 5,
+          privacy: 4,
+          respect: 5,
+          date_added: new Date(),
+          flagged: false,
+          flagged_reason: '',
+          admin_approved: true,
+          admin_edited: false,
+        },
+        {
+          id: 2,
+          landlord: 'Jane Smith',
+          country_code: 'US',
+          city: 'San Francisco',
+          state: 'CA',
+          zip: '94101',
+          review: 'I love this product!',
+          repair: 5,
+          health: 4,
+          stability: 5,
+          privacy: 3,
+          respect: 5,
+          date_added: new Date(),
+          flagged: false,
+          flagged_reason: '',
+          admin_approved: true,
+          admin_edited: false,
+        },
+      ];
+      const reviewUserSubmitted = 'This is an awesome product!';
+
+      const result = await service.checkRreviewsForSimilarity(
+        reviewsFromDb,
+        reviewUserSubmitted,
+      );
+
+      expect(result).toBe(true);
+    });
+
+    it('should return false if no similar review is found in the provided array', async () => {
+      const reviewsFromDb: Review[] = [
+        {
+          id: 1,
+          landlord: 'John Doe',
+          country_code: 'US',
+          city: 'New York',
+          state: 'NY',
+          zip: '10001',
+          review: 'This is a great product!',
+          repair: 4,
+          health: 3,
+          stability: 5,
+          privacy: 4,
+          respect: 5,
+          date_added: new Date(),
+          flagged: false,
+          flagged_reason: '',
+          admin_approved: true,
+          admin_edited: false,
+        },
+        {
+          id: 2,
+          landlord: 'Jane Smith',
+          country_code: 'US',
+          city: 'San Francisco',
+          state: 'CA',
+          zip: '94101',
+          review: 'I love this product!',
+          repair: 5,
+          health: 4,
+          stability: 5,
+          privacy: 3,
+          respect: 5,
+          date_added: new Date(),
+          flagged: false,
+          flagged_reason: '',
+          admin_approved: true,
+          admin_edited: false,
+        },
+      ];
+      const reviewUserSubmitted = 'This product is okay.';
+
+      const result = await service.checkRreviewsForSimilarity(
+        reviewsFromDb,
+        reviewUserSubmitted,
+      );
+
+      expect(result).toBe(false);
+    });
+  });
+});
```

Comment:
Working unit tests for my new ReviewSimilarityService who's purpose is to check if the review text coming back from the Database is too similar to the review text a person submits for a landlord in a new review.

Reply:
![image](https://github.com/RateTheLandlord/RateTheLandlord/assets/47609209/94244270-4b38-4610-b4a9-fc632a85ece7)